common4j/src/main/com/microsoft/identity/common/java/commands/DeviceCodeFlowTokenResultCommand.java

```diff
@@ -0,0 +1,122 @@
+import com.microsoft.identity.common.java.result.AcquireTokenResult;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.context.Scope;
+import lombok.NonNull;
+
+/**
+ * This command is used in the Device Code Flow (DCF) protocol to acquire token.
+ * Takes in a parameters object containing the  desired access scopes along with authorizationResult {@link AuthorizationResult}
+ * and returns a token result.
+ */
+public class DeviceCodeFlowTokenResultCommand extends TokenCommand{
+    private static final String TAG = DeviceCodeFlowTokenResultCommand.class.getSimpleName();
+
+    private final AuthorizationResult mAuthorizationResult;
+
+    public DeviceCodeFlowTokenResultCommand(@NonNull DeviceCodeFlowCommandParameters parameters,
+                                @NonNull AuthorizationResult authorizationResult,
+                                @NonNull BaseController controller,
+                                @SuppressWarnings(WarningType.rawtype_warning) @NonNull CommandCallback callback,
+                                @NonNull String publicApiId) {
+        super(parameters, controller, callback, publicApiId);
+
+        mAuthorizationResult = authorizationResult;
+    }
+
+    @Override
+    public AcquireTokenResult execute() throws Exception {
+        final String methodTag = TAG + ":execute";
+        Logger.verbose(
+                methodTag,
+                "DeviceCodeFlowTokenResultCommand initiating..."
+        );
+
+        final Span span = OTelUtility.createSpanFromParent(
+                SpanName.AcquireTokenDcf.name(), getParameters().getSpanContext()
+        );
+        span.setAttribute(AttributeName.application_name.name(), getParameters().getApplicationName());
+        span.setAttribute(AttributeName.public_api_id.name(), getPublicApiId());
+
+        try (final Scope scope = span.makeCurrent()) {
+            // Get the controller used to execute the command
+            final BaseController controller = getDefaultController();
+
+            // Fetch the parameters
+            final DeviceCodeFlowCommandParameters commandParameters = (DeviceCodeFlowCommandParameters) getParameters();
+
+            // Call acquireDeviceCodeFlowToken to get token result (Part 2 of DCF)
+            final AcquireTokenResult tokenResult = controller.acquireDeviceCodeFlowToken(mAuthorizationResult, commandParameters);
+
+            if (tokenResult == null) {
+                span.setStatus(StatusCode.ERROR, "empty result");
+            } else if (tokenResult.getSucceeded()) {
+                span.setStatus(StatusCode.OK);
+            } else {
+                final BaseException exception = ExceptionAdapter.exceptionFromAcquireTokenResult(tokenResult, getParameters());
+                if (!(exception.getErrorCode().equals(ErrorStrings.DEVICE_CODE_FLOW_AUTHORIZATION_PENDING_ERROR_CODE))) {
+                    if (exception != null) {
```

Comment:
This if exception != null should be outside instead of being nested inside if (exception.getErrorCode()...) block.. If exception is null, it would throw a null pointer here itself